addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -399,6 +402,9 @@ var SnippetEditor = Widget.extend({
      */
     toggleOverlayVisibility: function (show) {
         if (this.$el && !this.scrollingTimeout) {
+            if (!this._isInViewport(this.$target.get(0))) {
+                show = false;
+            }
```

Comment:
I am not sure to get the point of this code, can you explain? We may enabled snippets out of viewport via code then if the user scroll, the overlay should be ok... right?

Reply:
@bvr-odoo found out that sometimes snippets are "hidden" out of the viewport, which is now noticeable with my fix for the handle buttons, since when a snippet is hidden above the viewport we now see the handles dangling from above.